var/spack/repos/builtin/packages/sperr/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Sperr(CMakePackage):
+    """SPERR is a lossy scientific (floating-point) data compressor that can
+    perform either error-bounded or size-bounded data compression"""
+
+    homepage = "https://github.com/NCAR/SPERR"
+    git = "https://github.com/robertu94/SPERR"
```

Comment:
If your git repo is used as a staging area for changes that you eventually intend to upstream to https://github.com/NCAR/SPERR, then I think this should refer to that (elsewhere I suggest that you can create an inherited package if you have bugfixes that you need to use in the meantime - I'm curious if that would work here as well).

I'm curious if other packages following this theme (e.g. `sz-cpp`, `qoz`, `libpressio`, and `mgardx`, the last of which is actually a fork of a fork) can do this as well.

Reply:
Working with @shaomeng has been good and he's pretty quickly adopted fixes for spack.  We can probably switch over to his upstream.

`sz-cpp` became SZ3 but is different enough that some people want to compare against it separately.  They have different APIs so can't really be treated as the same.

`qoz` should eventually get merged into SZ3, but I am not sure of the timeline on this.

`libpressio` the development version is `robertu94` and the stable version is `codarcode`  I update the latter about 1/year.  `robertu94` gets small updates ~monthly.

`mgardx` was a research code based on MGARD with a different API.  Could someone (maybe  @qliu21) confirm if it will be merged into MGARD?  For now its a seperate package.

Reply:
@robertu94 I'm the main developer of `mgardx`. `mgardx` and the original `mgard` are in the same repository but they have two sets of APIs. `mgard` is the CPU-only version and `mgardx` is the CPU-GPU portable version, which make their internal design significantly different. In the foreseeable future,  we have no plan to merge them together.  For the propose of optimizing for Exascale computing, we are using `mgardx` as the default version and its API is used to integrate with ADIOS2.

Reply:
Has there been a release since this has happened? I can update LibPressio accordingly if it has

Reply:
For SPERR, I'm planning to put out a release in 2-3 weeks. Maybe I'll ping you when I do so you can update LibPressio?

Reply:
@shaomeng Sure. Are there api changes, or just internal changes?

Reply:
> internal c

There will not be API change, just internal tweaking, for the best I can tell. I can sync up with you again to run tests locally to make sure libpressio works well.

Reply:
@scheibelp I converted as many packages as practical to use upstream sources, and I pinned the versions.  Any package with a `libpressio-*` name is a package for which I am the main developer so packages with this prefix should have a `robertu94` git URL.  There are a few more remaining for which using upstream repos are not practical for reasons we discussed on the Spack call.  If the developers of these packages want to take over maintenance and provide stability under an open source licence, I'm happy to re-direct these remaining packages to point to their branches in the future.